serdes/generic/serde-common/src/main/java/io/apicurio/registry/serde/AbstractSerializer.java

```diff
@@ -0,0 +1,63 @@
+package io.apicurio.registry.serde;
+
+import io.apicurio.registry.resolver.ParsedSchema;
+import io.apicurio.registry.resolver.SchemaLookupResult;
+import io.apicurio.registry.resolver.SchemaResolver;
+import io.apicurio.registry.resolver.strategy.ArtifactReferenceResolverStrategy;
+import io.apicurio.registry.rest.client.RegistryClient;
+import io.apicurio.registry.serde.data.SerdeMetadata;
+import io.apicurio.registry.serde.data.SerdeRecord;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+
+public abstract class AbstractSerializer<T, U> extends AbstractSerDe<T, U> {
+
+    public AbstractSerializer() {
+        super();
+    }
+
+    public AbstractSerializer(RegistryClient client) {
+        super(client);
+    }
+
+    public AbstractSerializer(SchemaResolver<T, U> schemaResolver) {
+        super(schemaResolver);
+    }
+
+    public AbstractSerializer(RegistryClient client, SchemaResolver<T, U> schemaResolver) {
+        super(client, schemaResolver);
+    }
+
+    public AbstractSerializer(RegistryClient client, ArtifactReferenceResolverStrategy<T, U> strategy,
+            SchemaResolver<T, U> schemaResolver) {
+        super(client, strategy, schemaResolver);
+    }
+
+    protected abstract void serializeData(ParsedSchema<T> schema, U data, OutputStream out)
+            throws IOException;
+
+    public byte[] serializeData(String topic, U data) {
+        // just return null
+        if (data == null) {
+            return null;
+        }
+        try {
+            SerdeMetadata resolverMetadata = new SerdeMetadata(topic, isKey());
```

Comment:
Concepts of "topic" and "key" are somewhat Kafka specific, no?

Reply:
No, other messaging platforms have the concept of topic as well, like [Pulsar](https://pulsar.apache.org/docs/next/tutorials-topic/).